complex_functions/tests/jaccard_similarity_test.py

```diff
@@ -0,0 +1,76 @@
+import sys
+sys.path.append('../')
+from jaccard_similarity import jaccard_similariy
+
+# tests the algorithm on 2 sets
+def test_two_sets_False():
+    setA = {1}
+    setB = {0}
+    assert jaccard_similariy(setA, setB) == 0
+    assert jaccard_similariy(setB, setA) == 0
+
+# tests the algorithm on 2 sets with alternative union
+def test_two_sets_True():
+    setA = {1}
+    setB = {0}
+    assert jaccard_similariy(setA, setB, True) == 0
+    assert jaccard_similariy(setB, setA, True) == 0
+
+# tests the algorithm on 2 lists or tuples
+def test_two_lists_or_tuples_False():
+    setA = ['a', 'b', 'c', 'g']
+    setB = ('d', 'e', 'f', 'g')
+    assert jaccard_similariy(setA, setB) == 0.14285714285714285
+    # this test fails due to an uncovered case in the algorithm
+    # assert jaccard_similariy(setB, setA) == 0.14285714285714285
+
+# tests the algorithm on 2 sets or tuples with alternative union
+def test_two_lists_or_tuples_True():
+    setA = ('c', 'd', 'e', 'f', 'h', 'i')
+    setB = ('c', 'd', 'e', 'f', 'h', 'i')
+    # this test fails due to an uncovered case in the algorithm
+    # when the alternative union is chosen, union is a number
+    # and we try to do len(union) with is not valid
+    # assert jaccard_similariy(setA, setB, True) == 0
+    # assert jaccard_similariy(setB, setA, True) == 0
+
+# it tests all the invalid possible inputs
+# the commented assert are the ones where the right
+# behavior is not properly described in the doc
+def test_invalid_arguments():
+    assert jaccard_similariy({}, {}) == None
+    assert jaccard_similariy({}, {0}) == None
+    assert jaccard_similariy({0}, {}) == None
+    assert jaccard_similariy(0, 0) == None
+    assert jaccard_similariy({0}, 0) == None
+    assert jaccard_similariy(0, {0}) == None
+    assert jaccard_similariy(0, 0, True) == None
+    assert jaccard_similariy({0}, 0, True) == None
+    assert jaccard_similariy(0, {0}, True) == None
+    assert jaccard_similariy({}, {}, True) == None
+    assert jaccard_similariy({}, {0}, True) == None
+    assert jaccard_similariy({0}, {}, True) == None
+    # assert jaccard_similariy((), ()) == ?
+    # assert jaccard_similariy((), (0)) == ?
+    # assert jaccard_similariy((0), ()) == ?
+    assert jaccard_similariy(0, 0) == None
+    assert jaccard_similariy((0), 0) == None
+    assert jaccard_similariy(0, (0)) == None
+    assert jaccard_similariy(0, 0, True) == None
+    assert jaccard_similariy((0), 0, True) == None
+    assert jaccard_similariy(0, (0), True) == None
+    # assert jaccard_similariy((), (), True) == ?
+    # assert jaccard_similariy((), (0), True) == ?
+    # assert jaccard_similariy((0), (), True) == ?
+    # assert jaccard_similariy([], []) == ?
+    # assert jaccard_similariy([], [0]) == ?
+    # assert jaccard_similariy([0], []) == ?
```

Comment:
Can we remove these comments?

Reply:
As I mentioned above the test, the behavior of these cases aren't described in the doc, but if we follow the comprehension of the algorithm, this should be invalid inputs, but the algorithm seems to accept it. So what do you think ? Should I removed it or let it like I did ?

Reply:
I think it would look nicer to remove those comments.

Reply:
ok I'll do it